src/services/Store/BalanceService.tsx

```diff
@@ -199,15 +222,19 @@ export const getAccountsAssetsBalances = async (accounts: StoreAccount[]) => {
   return filteredUpdatedAccounts;
 };
 
-export const getAllTokensBalancesOfAccount = async (account: StoreAccount, assets: Asset[]) => {
-  const provider = new ProviderHandler(account.network);
-  const assetsInNetwork = assets.filter((x) => x.networkId === account.network.id);
+export const getAllTokensBalancesOfAccounts = async (
+  network: Network,
+  addresses: string[],
+  assets: Asset[]
+) => {
+  const provider = new ProviderHandler(network);
+  const assetsInNetwork = assets.filter((x) => x.networkId === network.id);
   const assetAddresses = getAssetAddresses(assetsInNetwork) as string[];
 
   try {
-    return tokenBalanceFetchWrapper(provider, account.address, assetAddresses, {
+    return tokenBalanceMultiFetchWrapper(provider, addresses, assetAddresses, {
```

Comment:
`return` without an `await` renders the catch useless.

```
let apiFail = () => new Promise((resolve, reject) => {
  reject('failed')
})

let run = async () => {
  try {
    console.log('try')
    return apiFail().then(() => console.log('then'))
  } catch (err) {
    console.log('catch', err)
  }
}

run()
```

<img width="740" alt="Screenshot 2020-12-14 at 15 39 46" src="https://user-images.githubusercontent.com/2429708/102094474-a734d600-3e22-11eb-8966-7eb5b736c91b.png">

Either `await` or use catch on the Promise.

Reply:
It can't error anyway as it is caught in the wrapper. So removing.